Out of bounds in `create_sem_set_with_id()`

### Describe the bug
There is a out of bounds error in `create_sem_set_with_id()` at kernel/aster-nix/src/ipc/semaphore/system_v/sem_set.rs:116 when make a `semget` syscall.

https://github.com/asterinas/asterinas/blob/0160a85ccd130c222ae9b8712b4102c726cba35c/kernel/aster-nix/src/ipc/semaphore/system_v/sem_set.rs#L116-L121

The out of bounds finally happens in `IdAlloc::alloc_specific()` at ostd/libs/id-alloc/src/lib.rs:160

https://github.com/asterinas/asterinas/blob/2102107be18729fba8f3546d5a3fe39c819d814a/ostd/libs/id-alloc/src/lib.rs#L156-L162

### To Reproduce
1. Compile a program which calls `semget`:
```C
#include <stdio.h>
#include <sys/ipc.h>
#include <sys/sem.h>
#include <errno.h>

int main() {
    semget(0x2209e1130c, 179, 0xaed34fff);

    return 0;
}

```
2. Run the compiled program in Asterinas.

### Expected behavior
Asterinas reports panic and is terminated.

### Environment
- Official docker asterinas/asterinas:0.6.2
- 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz
- Asterinas version: main 0160a85c

### Logs
```
~ # /root/semget.c
panicked at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/bitvec-1.0.1/src/slice/api.rs:2594:13:
index 165745420 out of bounds: 32001
Printing stack trace:
   1: fn 0xffffffff8878fae0 - pc 0xffffffff8878faf8 / registers:

     rax               0x12; rdx 0xffffffff88a22b80; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff8000014a9410;

   2: fn 0xffffffff8878f8c0 - pc 0xffffffff8878fa38 / registers:

     rax               0x12; rdx 0xffffffff88a22b80; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff8000014a9420;

   3: fn 0xffffffff88049000 - pc 0xffffffff8804900a / registers:

     rax               0x12; rdx 0xffffffff88a22b80; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff8000014a95a0;

   4: fn 0xffffffff889a30e0 - pc 0xffffffff889a3162 / registers:

     rax               0x12; rdx 0xffffffff88a22b80; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff8000014a95b0;

   5: fn 0xffffffff8893ea70 - pc 0xffffffff8893eb7f / registers:

     rax               0x12; rdx 0xffffffff88a22b80; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff8000014a9640;

   6: fn 0xffffffff8893f6a0 - pc 0xffffffff8893f70a / registers:

     rax               0x12; rdx 0xffffffff88a22b80; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff8000014a9720;

   7: fn 0xffffffff8893e9f0 - pc 0xffffffff8893ea5f / registers:

     rax               0x12; rdx 0xffffffff88a22b80; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff8000014a9780;

   8: fn 0xffffffff8893f410 - pc 0xffffffff8893f442 / registers:

     rax               0x12; rdx 0xffffffff88a22b80; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff8000014a97f0;

   9: fn 0xffffffff88939d40 - pc 0xffffffff88939daa / registers:

     rax               0x12; rdx 0xffffffff88a22b80; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff8000014a9860;

  10: fn 0xffffffff889422f0 - pc 0xffffffff88942314 / registers:

     rax               0x12; rdx 0xffffffff88a22b80; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff8000014a98c0;

  11: fn 0xffffffff883688f0 - pc 0xffffffff88368a79 / registers:

     rax               0x12; rdx 0xffffffff88a22b80; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff8000014a9920;

  12: fn 0xffffffff88444f80 - pc 0xffffffff884455c3 / registers:

     rax               0x12; rdx 0xffffffff88a22b80; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff8000014a9b90;

  13: fn 0xffffffff883a4760 - pc 0xffffffff883b8277 / registers:

     rax               0x12; rdx 0xffffffff88a22b80; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff8000014a9f30;

  14: fn 0xffffffff883a4480 - pc 0xffffffff883a450e / registers:

     rax               0x12; rdx 0xffffffff88a22b80; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff8000014c03d0;

  15: fn 0xffffffff8835e140 - pc 0xffffffff8835ecbf / registers:

     rax               0x12; rdx 0xffffffff88a22b80; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff8000014c0570;

  16: fn 0xffffffff885caf70 - pc 0xffffffff885caf7e / registers:

     rax               0x12; rdx 0xffffffff88a22b80; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff8000014c0f90;

  17: fn 0xffffffff887a5e30 - pc 0xffffffff887a5e46 / registers:

     rax               0x12; rdx 0xffffffff88a22b80; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff8000014c0fb0;

  18: fn 0xffffffff887dd9e0 - pc 0xffffffff887dda49 / registers:

     rax               0x12; rdx 0xffffffff88a22b80; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff8000014c0fd0;

  19: fn                0x0 - pc                0x0 / registers:

     rax               0x12; rdx 0xffffffff88a22b80; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff8000014c1000;

[OSDK] The kernel seems panicked. Parsing stack trace for source lines:
(  1) /root/asterinas/ostd/src/panicking.rs:107
(  2) /root/asterinas/ostd/src/panicking.rs:59
(  3) 2aghao2n2kcoquuybdyjuveav:?
(  4) ??:?
(  5) ??:?
(  6) /root/.rustup/toolchains/nightly-2024-06-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/option.rs:1006
(  7) /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/bitvec-1.0.1/src/slice/api.rs:2593
(  8) /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/bitvec-1.0.1/src/slice/ops.rs:159
(  9) /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/bitvec-1.0.1/src/vec/ops.rs:237
( 10) /root/asterinas/ostd/libs/id-alloc/src/lib.rs:160
( 11) /root/asterinas/kernel/aster-nix/src/ipc/semaphore/system_v/sem_set.rs:116
( 12) /root/asterinas/kernel/aster-nix/src/syscall/semget.rs:61
( 13) /root/asterinas/kernel/aster-nix/src/syscall/mod.rs:171
( 14) /root/asterinas/kernel/aster-nix/src/syscall/mod.rs:328
( 15) /root/asterinas/kernel/aster-nix/src/thread/task.rs:69
( 16) /root/.rustup/toolchains/nightly-2024-06-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/function.rs:79
( 17) /root/.rustup/toolchains/nightly-2024-06-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:2077
( 18) /root/asterinas/ostd/src/task/task/mod.rs:310
make: *** [Makefile:153: run] Error 1
```

--- a/kernel/src/ipc/semaphore/system_v/sem_set.rs
+++ b/kernel/src/ipc/semaphore/system_v/sem_set.rs
@@ -235,6 +235,9 @@ pub fn create_sem_set_with_id(
 ) -> Result<()> {
     debug_assert!(nsems <= SEMMSL);
     debug_assert!(id > 0);
+    if id as usize > SEMMNI {
+        return_errno_with_message!(Errno::ENOENT, "id larger than SEMMNI");
+    }
 
     ID_ALLOCATOR
         .get()

